File: server/db/dal/image_service.py
from db.dao.s3_dao import S3DAO

class ImageService:
    def __init__(self):
        self.s3_dao = S3DAO()

    def upload_image(self, file_path: str, image_name: str) -> str:

        object_name = f"2024/{image_name}"
        return self.s3_dao.upload_file(file_path, object_name)

    def download_image(self, image_name: str, download_path: str):
       
        object_name = f"images/{image_name}"
        self.s3_dao.download_file(object_name, download_path)

    def list_images(self) -> list:
        
        all_files = self.s3_dao.list_files()
 
        return [file for file in all_files if file.startswith("images/")]

File: server/db/dal/image_service.js
const S3DAO = require("../dao/s3_dao"); // Update the path based on your file structure

class ImageService {
    constructor() {
        this.s3DAO = new S3DAO();
    }

    /**
     * Upload an image to S3.
     * @param {string} filePath - Local file path.
     * @param {string} imageName - Desired name for the uploaded image.
     * @returns {Promise<string>} - Public URL of the uploaded image.
     */
    async uploadImage(bucketname, objectkey, file) {
        return this.s3DAO.uploadFile(bucketname, objectkey, file);
    }

    /**
     * Download an image from S3.
     * @param {string} imageName - S3 object key for the image.
     * @param {string} downloadPath - Local path to save the downloaded image.
     */
    async downloadImage(bucketname, objectName) {
        return await this.s3DAO.downloadFile(bucketname, objectName);
    }

    async deleteImage(bucketname, objectName) {
        return await this.s3DAO.deleteFile(bucketname, objectName);
    }
}

module.exports = ImageService;
